ui/src/OpenSearch.tsx

```diff
@@ -23,105 +23,145 @@ const getOpenSearchVersion = async () => {
     }
 }
 
-export const is2dot12plus = async () => {
-    const version = await getOpenSearchVersion();
-    console.log("opensearch version: " + version)
-    return semverGTE(version, "2.12.0");
-}
-
-export const hybridConversationSearchNoRag = async (rephrasedQuestion: string, index_name: string, model_id: string) => {
-    const query =
+export const hybridSearch = (rephrasedQuestion: string, filters: any, index_name: string, model_id: string) => {
+    let query =
     {
         "_source": SOURCES,
         "query": {
             "hybrid": {
                 "queries": [
                     {
-                        "match": {
-                            "text_representation": rephrasedQuestion
+                        "bool": {
+                            "must": [
+                                {
+                                    "exists": {
+                                        "field": "text_representation"
+                                    }
+                                },
+                                {
+                                    "match": {
+                                        "text_representation": rephrasedQuestion
+                                    }
+                                }
+                            ],
+                            "filter": [
+                                {
+                                    "match_all": {}
+                                }]
                         }
                     },
                     {
                         "neural": {
                             "embedding": {
                                 "query_text": rephrasedQuestion,
                                 "k": 100,
-                                "model_id": model_id
+                                "model_id": model_id,
+                                "filter":
+                                {
+                                    "match_all": {}
+                                }
                             }
                         }
                     }
                 ]
             }
         },
-        "size": 20
+        "size": 20,
+        // "ext": {
+        //     "rerank": {
+        //         "query_context": {
+        //             "query_text": rephrasedQuestion
+        //         }
+        //     }
+        // }
```

Comment:
maybe parametrize a `doReranking` field and conditionally add this like with filters?